src/main/java/com/mozilla/secops/gatekeeper/GuardDutyTransforms.java

```
@@ -112,9 +112,99 @@ public GenerateAlerts(Options opts) {
       }
     }
 
-    private void addEscalationMetadata(Alert a) {
+    private void addEscalationMetadata(Alert a, Finding f) {
       if (critNotifyEmail != null) {
-        a.addMetadata("notify_email_direct", critNotifyEmail);
+        for (Pattern p : escalate) {
+          if (p.matcher(f.getType()).matches()) {
+            a.addMetadata("notify_email_direct", critNotifyEmail);
+            return;
+          }
+        }
+      }
+    }
+
+    private void addLocalPortMetadata(Alert a, LocalPortDetails lpd) {
+      a.addMetadata("local_port", Integer.toString(lpd.getPort()));
```

Are these subfields always guaranteed to be non-null for example of `lpd` is non-null?

---

I was wondering if it would suffice (and be a good idea) to handle (not) setting a null value within the addMetadata() method?

---

Yeah we should probably have it throw an exception or something, but I don't think we would want to silently ignore that, we would still want to check these values if they are important enough to add.

---

As I commented here - https://github.com/mozilla-services/foxsec-pipeline/pull/203#issuecomment-516548528 - if `lpd.getPort()` returns `null`, `Integer.toString(lpd.getPort())` will throw a null pointer exception.